module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -509,45 +517,64 @@ namespace module::input {
+
+                        // Calculate values needed for Z_HEIGHT method
+                        const Eigen::Affine3d Htr(sensors->Htx[ServoID::R_ANKLE_ROLL]);
+                        const Eigen::Affine3d Htl(sensors->Htx[ServoID::L_ANKLE_ROLL]);
+                        const Eigen::Affine3d Hlr  = Htl.inverse() * Htr;
+                        const Eigen::Vector3d rRLl = Hlr.translation();
+
+                        switch (config.footDown.method()) {
+                            case FootDownMethod::LOAD:
+                                // Use our virtual load sensor class to work out which feet are down
+                                feet_down = load_sensor.updateFeet(*sensors);
+                                break;
+                            case FootDownMethod::Z_HEIGHT:
+                                // Right foot is below left foot in left foot space
+                                if (rRLl.z() < -config.footDown.threshold()) {
+                                    feet_down[BodySide::RIGHT] = true;
+                                    feet_down[BodySide::LEFT]  = false;
+                                }
+                                // Right foot is above left foot in left foot space
+                                else if (rRLl.z() > config.footDown.threshold()) {
+                                    feet_down[BodySide::RIGHT] = false;
+                                    feet_down[BodySide::LEFT]  = true;
+                                }
+                                // Right foot and left foot are roughly the same height in left foot space
+                                else {
+                                    feet_down[BodySide::RIGHT] = true;
+                                    feet_down[BodySide::LEFT]  = true;
+                                }
+                                break;
+                            case FootDownMethod::FSR:
+                                // For a foot to be on the ground we want a minimum of 2 diagonally opposite studs
+                                // in contact with the ground
+                                // So fsr1 and fsr3, or fsr2 and fsr4
+                                //
+                                // A FSR is in contact with the ground if its value is greater than the certainty
+                                // threshold
+
+                                feet_down[BodySide::LEFT] =
+                                    (((input.fsr.left.fsr1 > config.footDown.threshold())
+                                      && (input.fsr.left.fsr3 > config.footDown.threshold()))
+                                     || ((input.fsr.left.fsr2 > config.footDown.threshold())
+                                         && (input.fsr.left.fsr4 > config.footDown.threshold())));
+
+                                feet_down[BodySide::RIGHT] =
+                                    (((input.fsr.right.fsr1 > config.footDown.threshold())
```

Comment:
Just confirming - when the stud is pressed in, the value of this will be large? (or at least larger than the threshold)

Reply:
Ye